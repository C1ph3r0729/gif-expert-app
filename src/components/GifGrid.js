import React from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data: gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn"> {category}</h3>
      {loading && <p>Loading ...</p>}
      <div className="card-grid">
        {gifs.map((gif) => (
          <GifGridItem {...gif} key={gif.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
