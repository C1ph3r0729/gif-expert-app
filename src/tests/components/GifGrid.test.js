import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("<GifGrid />", () => {
  const category = "Rick and Morty";
  // let wrapper = shallow(<GifGrid category={category} />);

  test("should display the component correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should display items when loading images from useFetchGifs", () => {
    const gifs = [
      {
        id: "dadadsda",
        url: "https://localhost/algo/img.jpg",
        title: "Rick and Morty in the space",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
