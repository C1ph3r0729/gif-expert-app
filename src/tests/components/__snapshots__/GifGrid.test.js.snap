// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> should display items when loading images from useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Rick and Morty
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="dadadsda"
      key="dadadsda"
      title="Rick and Morty in the space"
      url="https://localhost/algo/img.jpg"
    />
  </div>
</Fragment>
`;

exports[`<GifGrid /> should display the component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Rick and Morty
  </h3>
  <p>
    Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
